### ISSUE TITLE
Upgrade to Babel 7

### ISSUE THREAD

**Issue by jonathanmoore** [Shopify/slate#931]
### Problem

Babel 7 introduced a lot of new, useful functionality, but most of all I would like to take advantage of `useBuiltIns: 'usage'` to help with polyfills.  Currently, with Slate you either have to manually keep up with polyfills one by one as you develop or take the shotgun approach of importing all polyfills for the target browsers regardless of usage.
https://babeljs.io/docs/en/babel-preset-env#usebuiltins-usage-experimental

With this new approach, Babel will automatically look for code that needs a polyfill and import the correct modules.
```
// In
var a = new Promise();

// Output
import "core-js/modules/es6.promise";
var a = new Promise();
```

### More Information

I did take an initial attempt at migrating to Babel 7 locally, and after updating `slate-tools` the polyfills were loading as expected.  However, I was running into additional conflicts with [Shopify/babel-preset-shopify](https://github.com/Shopify/babel-preset-shopify) in the `.babelrc` file as well as live-reloading. 

Upgrading will probably be a big effort, but it will help tremendously with development and even `theme-scripts` where a developer might assume the browser support is already there. For example:
https://github.com/Shopify/theme-scripts/blob/43383342be21ab42b5fe4c78be98ec3655279ef5/packages/theme-cart/README.md#browser-support

**Comment by justinmetros** [comment]
[babel-preset-shopify](https://github.com/Shopify/babel-preset-shopify/issues/31) v17.0.0 was just released w/ Babel version 7.  

Also tried a local move to Babel but slate-tools wasn't liking it:
```
ERROR in ./scripts/templates/index.js
Module build failed (from ../node_modules/@shopify/slate-tools/node_modules/babel-loader/lib/index.js):
```

**Comment by macrozone** [comment]
yes slate-tools has dependencies on old babel version and should be upgraded.

**Comment by uharbachou** [comment]
hi, Any updates with upgrade packages ?

**Comment by ryaninvents** [comment]
@macrozone do you have an example of a working Storybook configuration? I'm in the exact same situation
